File: sdks/guides/typescript/multi-agent/mock-llm.ts
let orchestratorCallCount = 0;

export interface ToolCallResponse {
  done: boolean;
  content: string;
  toolCall?: { name: string; args: { task: string } };
}

export function mockOrchestratorLlm(messages: Array<{ role: string; content: string }>): ToolCallResponse {
  orchestratorCallCount++;
  switch (orchestratorCallCount) {
    case 1:
      return { done: false, content: '', toolCall: { name: 'research', args: { task: 'Find key facts about the topic' } } };
    case 2:
      return { done: false, content: '', toolCall: { name: 'writing', args: { task: 'Write a summary from the research' } } };
    default:
      return { done: true, content: 'Here is the final report combining research and writing.' };
  }
}

export function mockSpecialistLlm(task: string, role: string): string {
  return `[${role}] Completed: ${task}`;
}

File: examples/go/guides/multi-agent/mock_llm.go
package main

import "fmt"

var orchestratorCallCount int

type OrchestratorResponse struct {
	Done     bool
	Content  string
	ToolCall *struct {
		Name string
		Args map[string]string
	}
}

func MockOrchestratorLLM(messages []map[string]interface{}) OrchestratorResponse {
	orchestratorCallCount++
	switch orchestratorCallCount {
	case 1:
		return OrchestratorResponse{Done: false, ToolCall: &struct {
			Name string
			Args map[string]string
		}{Name: "research", Args: map[string]string{"task": "Find key facts about the topic"}}}
	case 2:
		return OrchestratorResponse{Done: false, ToolCall: &struct {
			Name string
			Args map[string]string
		}{Name: "writing", Args: map[string]string{"task": "Write a summary from the research"}}}
	default:
		return OrchestratorResponse{Done: true, Content: "Here is the final report combining research and writing."}
	}
}

func MockSpecialistLLM(task, role string) string {
	return fmt.Sprintf("[%s] Completed: %s", role, task)
}
